c++/src/kj/memory.h

```diff
@@ -25,6 +25,7 @@
 #ifdef KJ_DEBUG
 #include <atomic>  // std::atomic for KJ_ASSERT_PTR_COUNTERS
 #endif
+#include <cstddef>
```

Comment:
Changed the include. I assume we also want to use `nullptr_t` instead of `std::nullptr_t` then.

Reply:
`nullptr_t` is C++-specific, not C. The C++ standard library is indeed scoped to `std::`, but I've honestly never seen anyone use the `std::` prefix on C library declarations.

(Though FWIW I usually use `decltype(nullptr)` instead of `nullptr_t` so that I don't have to include a header at all.)

Reply:
Wait now I'm confused.

I missed that the reason you were adding stddef is for `nullptr_t`... I'm surprised that `nullptr_t` is in stddef since it's not C. But indeed looking at the docs, it says it's in `<cstddef>`? https://en.cppreference.com/w/cpp/types/nullptr_t

I was under the impression that `cstddef` contained _strictly_ the contents of C's `stddef.h` and not any C++-specific stuff, but apparently I am wrong.

Reply:
Could we just change any uses of `nullptr_t` to `decltype(nullptr)` and sidestep the issue?

I have always used `decltype(nullptr)` before, any use of `nullptr_t` must have been added by someone else.

Reply:
And actually, I definitely don't want to add any unnecessary headers to `memory.h`, which is a KJ file that basically everyone includes. So `decltype(nullptr)` is really the best solution here.

Reply:
Updated to use decltype(nullptr).